### PR TITLE
samples: nrf_desktop: DFU helper commands

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-524

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

File: scripts/hid_configurator/configurator.py [hunk 1]
```diff
@@ -35,14 +36,64 @@
...
+    SUCCESS            = 0
+    PENDING            = 1
+    FETCH              = 2
+    TIMEOUT            = 3
+    REJECT             = 4
+    WRITE_ERROR        = 5
     DISCONNECTED_ERROR = 6
+    FAULT              = 99
+
+class Response(object):
```
Comment (Qbicz) [inline scripts/hid_configurator/configurator.py]: +1 for encapsulating the response handling

File: scripts/hid_configurator/configurator.py [hunk 2]
```diff
@@ -249,12 +304,56 @@ def perform_config(dev, args):
...
+    event_id = (EVENT_GROUP_DFU << GROUP_FIELD_POS) | (DFU_IMGINFO << TYPE_FIELD_POS)
+    recipient = MOUSE_PID
+
+    try:
+        success, fetched_data = exchange_feature_report(dev, recipient, event_id, None, True)
+    except:
+        success = False
+
+    if success and fetched_data:
+        fmt = '<BIBBHI'
```
Comment (Qbicz) [inline scripts/hid_configurator/configurator.py]: assert(struct.calcsize(fmt) <= EVENT_DATA_LEN_MAX)

Comment (pdunaj) [reply scripts/hid_configurator/configurator.py]: added

File: scripts/hid_configurator/configurator.py [hunk 3]
```diff
@@ -108,52 +159,51 @@ def create_fetch_report(recipient, event_id):
...
-        response = dev.get_feature_report(REPORT_ID, REPORT_SIZE)
 
-        resp_recipient, resp_event_id, resp_status, resp_len, resp_data = parse_report(response)
-        if is_fetch:
-            print('Fetched value:', int.from_bytes(resp_data, byteorder='little'))
+        response_raw = dev.get_feature_report(REPORT_ID, REPORT_SIZE)
+        response = Response.parse_response(response_raw)
 
-        if resp_recipient != recipient or resp_event_id != event_id:
+        if not response:
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply scripts/hid_configurator/configurator.py]: Fixed

File: applications/nrf_desktop/src/modules/dfu.c [hunk 4]
```diff
@@ -185,16 +204,131 @@ static void handle_dfu_data(u8_t *data, size_t size)
...
+
+		sys_put_le16(header.h.v1.sem_ver.revision, &fetch_event->dyndata.data[pos]);
+		pos += sizeof(header.h.v1.sem_ver.revision);
+
+		sys_put_le32(header.h.v1.sem_ver.build_num, &fetch_event->dyndata.data[pos]);
+		pos += sizeof(header.h.v1.sem_ver.build_num);
+
+		EVENT_SUBMIT(fetch_event);
+	} else {
+		LOG_WRN("Cannot obtain image information");
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply applications/nrf_desktop/src/modules/dfu.c]: Ok

File: applications/nrf_desktop/src/modules/dfu.c [hunk 5]
```diff
@@ -185,16 +204,131 @@ static void handle_dfu_data(u8_t *data, size_t size)
 	k_delayed_work_cancel(&dfu_timeout);
 }
 
+static void handle_reboot_request(const struct config_fetch_request_event *event)
+{
+	LOG_INF("System reboot requested");
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply applications/nrf_desktop/src/modules/dfu.c]: Rephrased

File: applications/nrf_desktop/src/modules/dfu.c [hunk 6]
```diff
@@ -87,8 +89,24 @@ static void dfu_timeout_handler(struct k_work *work)
 	}
 }
 
-static void handle_dfu_data(u8_t *data, size_t size)
+static void reboot_request_handler(struct k_work *work)
 {
+	LOG_INF("Handle reset requested now");
```
Comment (Qbicz) [inline applications/nrf_desktop/src/modules/dfu.c]: requested -> request

File: applications/nrf_desktop/src/modules/dfu.c [hunk 7]
```diff
@@ -134,7 +152,8 @@ static void handle_dfu_data(u8_t *data, size_t size)
 	}
 
 	if (flash_area->fa_size < img_length) {
-		LOG_WRN("Insufficient space for DFU");
+		LOG_WRN("Insufficient space for DFU (%zu < %zu)",
+			(size_t)flash_area->fa_size, (size_t)img_length);
```
Comment (Qbicz) [inline applications/nrf_desktop/src/modules/dfu.c]: Why cast both u32 to size_t?

Comment (Qbicz) [reply applications/nrf_desktop/src/modules/dfu.c]: You could use "%u" or "%" PRIu32

Comment (pdunaj) [reply applications/nrf_desktop/src/modules/dfu.c]: Fixed

File: applications/nrf_desktop/src/modules/dfu.c (hunk 6 above)
Comment (pdunaj) [reply applications/nrf_desktop/src/modules/dfu.c]: Fixed

**Comment by pdunaj** [comment]
@thst-nordic , @wiminordic does you know what is happening with the last CI test? It seems this is something new (previously there was only 12 tests) and it does not finish.

**Comment by pdunaj** [comment]
HI @pizi-nordic , could you merge?

**Pull request merged**